sympy/concrete/summations.py

```diff
@@ -1098,3 +1099,35 @@ def eval_sum_hyper(f, i_a_b):
                     return S.NegativeInfinity
             return None
         return Piecewise(res, (old_sum, True))
+
+def asymptotic(expr, symbol):
+    '''Tries to perform the asymptotic of expr when
+    symbols tends to infinity
+
+    return either the asymptotic expression or the original one
+    '''
+    from sympy import simplify
+    if type(expr) is Mul:
+        # do the asymptotic for each factor
+        tmp = [asymptotic(i, symbol) for i in expr.args]
+        result = S(1)
+        for i in tmp:
+            result = result * i
+        return result
+
+    try:
+        expr = simplify(expr)
+        lim = limit(expr, symbol, S.Infinity)
+        if lim == S.Infinity:
+            return expr
+        if lim != S(0): # the asymptotic is the value of the limit
+            return lim
+        expr2 = expr.subs(symbol, 1/symbol)
+        s = expr2.series(symbol, 0, 10) # use MacLaurin expansion
+        if type(s) is Add:
+            s = s.subs(symbol, 1/symbol)
+            return s.args[0]
+        else:
+            return expr
+    except:
```

Comment:
I put `except Exception:`. I am not sure if it is ok. I tried with `except NotImplementedError:`, but some tests failed.